OpenSim/Simulation/Test/testSimulationUtilities.cpp

```diff
@@ -0,0 +1,151 @@
+                updatedKinematics->getStateIndex("knee_angle_pat_r"),
+                SimTK_RADIAN_TO_DEGREE);
+        
+        const int numColumns = origKinematics.getColumnLabels().getSize();
+        CHECK_STORAGE_AGAINST_STANDARD(*updatedKinematics, origKinematics,
+                std::vector<double>(numColumns, 1e-14),
+                __FILE__, __LINE__,
+                "updatePre40KinematicsStorageFor40MotionType() altered columms incorrectly.");
+    }
+    
+    // Test updatePre40KinematicsFilesFor40MotionType().
+    const std::string updatedKinematicsFile =
+            "testSimulationUtilities_leg69_IK_stance_pre4_updated.mot";
+    if (IO::FileExists(updatedKinematicsFile)) {
+        std::remove(updatedKinematicsFile.c_str());
+    }
+    updatePre40KinematicsFilesFor40MotionType(model, {origKinematicsFile});
+    SimTK_TEST(IO::FileExists(updatedKinematicsFile));
+    
+    {
+        Storage updatedKinematics(updatedKinematicsFile);
+        updatedKinematics.multiplyColumn(
+            updatedKinematics.getStateIndex("knee_angle_pat_r"),
+            SimTK_RADIAN_TO_DEGREE);
+    
+        const int numColumns = origKinematics.getColumnLabels().getSize();
+        // We lose precision when through the file.
+        CHECK_STORAGE_AGAINST_STANDARD(updatedKinematics, origKinematics,
+                std::vector<double>(numColumns, 1e-6),
+                __FILE__, __LINE__,
+                "updatePre40KinematicsFilesFor40MotionType() altered columms incorrectly.");
+    }
+    
+    {
+        // Check that we can still update data with a copy of the model
+        // (even though the XML document is gone).
+        Model modelCopy(model);
+        SimTK_TEST(!model.getWarningMesssageForMotionTypeInconsistency().empty());
+        
+        auto updatedKinematics =
+            updatePre40KinematicsStorageFor40MotionType(model, origKinematics);
+        
+        updatedKinematics->multiplyColumn(
+                updatedKinematics->getStateIndex("knee_angle_pat_r"),
+                SimTK_RADIAN_TO_DEGREE);
+        
+        const int numColumns = origKinematics.getColumnLabels().getSize();
+        CHECK_STORAGE_AGAINST_STANDARD(*updatedKinematics, origKinematics,
+                std::vector<double>(numColumns, 1e-14),
+                __FILE__, __LINE__,
+                "updatePre40KinematicsStorageFor40MotionType(), with a copied "
+                "model, altered columms incorrectly.");
+    }
+    const std::string updatedModelFile =
+            "testSimulationUtilities_leg6dof9musc_updated.osim";
+    if (IO::FileExists(updatedModelFile)) {
+        std::remove(updatedModelFile.c_str());
+    }
+    {
+        // TODO check that we get the exception if a model has no document version.
+        model.print(updatedModelFile);
+        Model updatedModel(updatedModelFile);
+        SimTK_TEST(updatedModel.getWarningMesssageForMotionTypeInconsistency().empty());
+        
+        SimTK_TEST_MUST_THROW_EXC(
```

Comment:
What is the exception you are expecting here? Method "has no updates to make"? Perhaps state that in a comment. 
I realize now the previous comment is a TODO but I was expecting the exception to be related to no document version.